pkg/utils/controller/controller_test.go

```diff
@@ -128,6 +128,18 @@ func TestNewAPIServerError(t *testing.T) {
 			err:       apierrors.NewConflict(schema.GroupResource{}, "conflict", nil),
 			wantErr:   ErrAPIServerError,
 		},
+		{
```

Comment:
add these two cases when the fromCache is true

add a case when fromCache is true and input err: runtime.NewMissingKindErr("unstructured object has no kind"), which is possible when reading from the cache